### PR TITLE
[Shared] 1.4 - Action.Execute

### LINKED ISSUE #5500: [Shared] Support parsing of Action.Execute

### PR DESCRIPTION
**Pull request by golddove**
## Related Issue
Fixes #5500

Note: did not re-swig, will include that in the Android part

## Description
Added `Refresh`, `Authentication`, and `Action.Execute` types to object model

## Sample Card
`EverythingBagel.json` in shared model tests dir

## How Verified
Added properties to EverythingBagel

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5539)

### CODE REVIEW

**Review by dclaux — APPROVED** [review]
Looks good to me.

File: source/shared/cpp/ObjectModel/Refresh.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "pch.h"
+#include "BaseActionElement.h"
+
+namespace AdaptiveSharedNamespace
+{
+    class Refresh
+    {
+    public:
+        Refresh() : Refresh(nullptr, {}) {}
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/Refresh.h:13]:
```suggestion
        Refresh() {}
```

**Review by paulcam206 — APPROVED** [review]
minor nits only -- looks good!

**Comment by paulcam206** [comment]
if this is ready, let's get this in to unblock UWP :)

**Comment by paulcam206** [comment]
@golddove - looks like there are some sharedmodel test failures that need to be addressed...

**Comment by golddove** [comment]
Merging for now. Android/iOS build issues shall be fixed in those respective PR's.

**Pull request merged**